gulpfile.js

```diff
@@ -225,19 +261,36 @@ gulp.task('autoupdate:push:win', function (done) {
       CacheControl: 'public, max-age=0, no-cache',
       ACL: 'public-read'
     }
-  }, done);
+  });
 });
 
-gulp.task('manifest:push', function (done) {
-  pushS3({
+
+
+var pushManifestToDest = function(destinationKey) {
+  return pushS3({
     localFile: SOURCE_DIR + '/package.json',
     s3Params: {
       Bucket: S3_CONSTS.buckets.updates,
-      Key: 'desktop/package.json',
+      Key: destinationKey,
       CacheControl: 'public, max-age=0, no-cache',
       ACL: 'public-read'
     }
-  }, done);
+  });
+};
+var platformFolders = [
+  // legacy
+  'desktop',
+  'osx',
+  'win',
+  'linux'
+];
+platformFolders.forEach(function(platformFolder) {
+  gulp.task('manifest:push:' + platformFolder, function() {
+    return pushManifestToDest(path.join(platformFolder, 'package.json'));
+  });
+});
```

Comment:
I did this based on existing patterns like `cert:fetch:xxx`, `clean:xxx`, `pack:templates:xxx`, `pack:deb:xxx`, `artefacts:push:xxx`, etc, etc